server/sonar-server/src/main/java/org/sonar/server/webhook/ws/UpdateAction.java

```diff
@@ -0,0 +1,128 @@
+import static org.sonar.server.ws.KeyExamples.URL_WEBHOOK_EXAMPLE_001;
+import static org.sonar.server.ws.WsUtils.checkFoundWithOptional;
+
+public class UpdateAction implements WebhooksWsAction {
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final WebhookSupport webhookSupport;
+
+  public UpdateAction(DbClient dbClient, UserSession userSession, WebhookSupport webhookSupport) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.webhookSupport = webhookSupport;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+
+    WebService.NewAction action = controller.createAction(UPDATE_ACTION)
+      .setPost(true)
+      .setDescription("Update a Webhook.<br>" +
+        "Requires the global, organization or project permission.")
+      .setSince("7.1")
+      .setResponseExample(getClass().getResource("example-webhook-create.json"))
+      .setHandler(this);
+
+    action.createParam(KEY_PARAM)
+      .setRequired(true)
+      .setMaximumLength(KEY_PARAM_MAXIMUN_LENGTH)
+      .setDescription("The key of the webhook to be updated")
+      .setExampleValue(KEY_PROJECT_EXAMPLE_001);
+
+    action.createParam(NAME_PARAM)
+      .setRequired(true)
+      .setMaximumLength(NAME_PARAM_MAXIMUM_LENGTH)
+      .setDescription("new name of the webhook")
+      .setExampleValue(NAME_WEBHOOK_EXAMPLE_001);
+
+    action.createParam(URL_PARAM)
+      .setRequired(true)
+      .setMaximumLength(URL_PARAM_MAXIMUM_LENGTH)
+      .setDescription("new url to be called by the webhook")
+      .setExampleValue(URL_WEBHOOK_EXAMPLE_001);
+
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+
+    userSession.checkLoggedIn();
+
+    String webhookKey = request.param(KEY_PARAM);
+    String name = request.mandatoryParam(NAME_PARAM);
+    String url = request.mandatoryParam(URL_PARAM);
+
+    webhookSupport.checkUrlPattern(url, "Url parameter with value '%s' is not a valid url", url);
+
+    try (DbSession dbSession = dbClient.openSession(false)) {
+
+      Optional<WebhookDto> dtoOptional = dbClient.webhookDao().selectByUuid(dbSession, webhookKey);
+      WebhookDto webhookDto = checkFoundWithOptional(dtoOptional, "No webhook with key '%s'", webhookKey);
+
+      String organizationUuid = webhookDto.getOrganizationUuid();
+      if (organizationUuid != null) {
+        Optional<OrganizationDto> optionalDto = dbClient.organizationDao().selectByUuid(dbSession, organizationUuid);
+        webhookSupport.checkUserPermissionOn(optionalDto.get());
+        updateWebhook(dbSession, webhookDto, name, url);
+      }
+
+      String projectUuid = webhookDto.getProjectUuid();
+      if (projectUuid != null) {
+        Optional<ComponentDto> optionalDto = Optional.ofNullable(dbClient.componentDao().selectByUuid(dbSession, projectUuid).orNull());
+        webhookSupport.checkUserPermissionOn(optionalDto.get());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "optionalDto.isPresent()" before accessing the value. ([squid:S3655](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10345_manage_webhooks&issues=AWF1LkN70orRW2HVTtYw&open=AWF1LkN70orRW2HVTtYw)

[ISSUE_KEY]: AWF1LkN70orRW2HVTtYw